backend/src/services/instance/instance.service.ts

```diff
@@ -108,27 +123,47 @@ class Instance {
       name,
       keyPairId,
       username: InstanceDefaultParam.USERNAME as string,
-      hostname: hostname,
+      hostname: '',
       operationSystemId,
       createdBy: userId,
       awsInstanceId: instanceId,
       tenantId,
+      state,
     });
 
-    await this.#instanceRepository.create(instance);
+    const { id } = await this.#instanceRepository.create(instance);
+
+    this.#eventService.on(EventName.ADD_NEW_INSTANCE, async ({ id, awsId }) => {
+      const { hostname, state } = await this.#ec2Service.getRunningInstanceData(
+        awsId,
+      );
+      await this.update(id, { hostname, state });
+    });
+
+    this.#eventService.emit(EventName.ADD_NEW_INSTANCE, {
+      id,
+      awsId: instanceId,
+    });
```

Comment:
Honestly, I do not completely understand what happens here. Are you trying to do some work in the background? If so, why not just use IIFE?
```
(async (): Promise<void> => {
      const { hostname, state } = await this.#ec2Service.getRunningInstanceData(
        instanceId,
      );
      await this.update(id, { hostname, state });
    })();